app/scripts/controllers/alert-controller.test.ts

```diff
@@ -28,231 +29,288 @@ const EMPTY_ACCOUNT = {
+): Promise<ReturnValue> {
+  const [{ ...rest }, fn] = args.length === 2 ? args : [{}, args[0]];
+  const { ...alertControllerOptions } = rest;
 
-    const alertMessenger = controllerMessenger.getRestricted({
+  const controllerMessenger = new ControllerMessenger<
+    AllowedActions | AlertControllerGetStateAction,
+    AllowedEvents | AlertControllerStateChangeEvent
+  >();
+
+  const alertControllerMessenger: AlertControllerMessenger =
+    controllerMessenger.getRestricted({
       name: 'AlertController',
-      allowedActions: [`AccountsController:getSelectedAccount`],
-      allowedEvents: [`AccountsController:selectedAccountChange`],
+      allowedActions: ['AccountsController:getSelectedAccount'],
+      allowedEvents: ['AccountsController:selectedAccountChange'],
     });
 
-    alertController = new AlertController({
-      state: {
-        unconnectedAccountAlertShownOrigins: {
-          testUnconnectedOrigin: false,
-        },
-        web3ShimUsageOrigins: {
-          testWeb3ShimUsageOrigin: 0,
-        },
-      },
-      controllerMessenger: alertMessenger,
-    });
+  controllerMessenger.registerActionHandler(
+    'AccountsController:getSelectedAccount',
+    jest.fn().mockReturnValue(EMPTY_ACCOUNT),
+  );
+
+  const controller = new AlertController({
+    messenger: alertControllerMessenger,
+    state: getDefaultAlertControllerState(),
+    ...alertControllerOptions,
+  });
+
+  return await fn({
+    controller,
+    messenger: controllerMessenger,
   });
+}
 
+describe('AlertController', () => {
   describe('default state', () => {
-    it('should be same as AlertControllerState initialized', () => {
-      expect(alertController.store.getState()).toStrictEqual({
-        alertEnabledness: {
-          unconnectedAccount: true,
-          web3ShimUsage: true,
-        },
-        unconnectedAccountAlertShownOrigins: {
-          testUnconnectedOrigin: false,
-        },
-        web3ShimUsageOrigins: {
-          testWeb3ShimUsageOrigin: 0,
-        },
+    it('should be same as AlertControllerState initialized', async () => {
+      await withController({ state: initialState }, ({ controller }) => {
+        expect(controller.state).toStrictEqual({
+          alertEnabledness: {
+            unconnectedAccount: true,
+            web3ShimUsage: true,
+          },
+          unconnectedAccountAlertShownOrigins: {
+            testUnconnectedOrigin: false,
+          },
+          web3ShimUsageOrigins: {
+            testWeb3ShimUsageOrigin: 0,
+          },
+        });
       });
     });
   });
 
   describe('alertEnabledness', () => {
-    it('should default unconnectedAccount of alertEnabledness to true', () => {
-      expect(
-        alertController.store.getState().alertEnabledness.unconnectedAccount,
-      ).toStrictEqual(true);
+    it('should default unconnectedAccount of alertEnabledness to true', async () => {
+      await withController({ state: initialState }, ({ controller }) => {
```

Comment:
Do we need to pass initial state? The controller is defined such the `state` option is optional, in which case it just uses the default state. So it seems like we could say:

``` suggestion
      await withController(({ controller }) => {